qb/test/params.test.ts

```diff
@@ -0,0 +1,36 @@
+import {typeutil} from "../../src/reflection";
+import e from "../generated/example";
+
+test("simple params", () => {
+  const query = e.withParams(
+    {
+      str: e.str,
+      numArr: e.array(e.int64),
+      optBool: e.optional(e.bool),
+    },
+    (params) =>
+      e.select({
+        str: params.str,
+        nums: e.array_unpack(params.numArr),
+        x: e.if_else(e.str("true"), params.optBool, e.str("false")),
+      })
+  );
+
+  expect(query.toEdgeQL()).toEqual(`SELECT {
+  str := (<std::str>$str),
+  nums := (std::array_unpack((<array<std::int64>>$numArr))),
+  x := (("true" IF <OPTIONAL std::bool>$optBool ELSE "false"))
+}`);
```

Comment:
I think that instead of testing the output of `toEdgeQL()` we should execute the queries against the server and check the result. The rendering of queries might change, but what the compile to semantically must be the same across all versions.

Reply:
IOW it's more cumbersome but better in the long term, I believe.